build.sbt

```diff
@@ -129,10 +130,36 @@ lazy val core = crossProject(JSPlatform, JVMPlatform, NativePlatform).crossType(
   .settings(mimaSettings:_*)
   .jsSettings(commonJsSettings:_*)
   .jvmSettings(commonJvmSettings:_*)
+  .nativeSettings(
+    // disable scaladoc generation on native
+    // currently getting errors like
+    //   [error] bnd: Invalid syntax for version: ${@}, for cmd: range, arguments; [range, [==,=+), ${@}]
+    publishArtifact in (Compile, packageDoc) := false,
+    publishArtifact in packageDoc := false,
+    sources in (Compile,doc) := Seq.empty
+  )
+
+val macroCompatUri = uri("https://github.com/milessabin/macro-compat.git#5b4fccdbfaf6bd730725f1cde0cbbd50bf4c3e78")
+lazy val coreJVM = core.jvm.dependsOn(
+  ProjectRef(
+    macroCompatUri,
+    "coreJVM"
+  )
+)
 
-lazy val coreJVM = core.jvm
-lazy val coreJS = core.js
-lazy val coreNative = core.native
+lazy val coreJS = core.js.dependsOn(
+  ProjectRef(
+    macroCompatUri,
+    "coreJS"
+  )
+)
+
+lazy val coreNative = core.native.dependsOn(
+  ProjectRef(
+    macroCompatUri,
+    "coreJVM"
```

Comment:
Is it right for the native cross-project to express a dependency on macro-compat's `coreJVM`? I'm not yet familiar with the how Scala native code is distributed so I want to confirm that this is intentional.

Reply:
Well, it's a workaround. It's how the current shapeless master appears to work (only in that case it's a JAR dependency, and here it's source). I think it works because (1) on 2.11 macro-compat is compile-time-only and (2) currently scala native is 2.11-only.

Reply:
Oh yeah! Make sense, since it's really a pure macro-space project.